wpilibj/src/main/java/edu/wpi/first/wpilibj/AddressableLED.java

```diff
@@ -8,48 +8,33 @@
 package edu.wpi.first.wpilibj;
 
 import edu.wpi.first.hal.AddressableLEDJNI;
+import edu.wpi.first.hal.HAL;
+import edu.wpi.first.hal.PWMJNI;
 
 /**
  * A class for driving addressable LEDs, such as WS2812s and NeoPixels.
  */
 public class AddressableLED implements AutoCloseable {
-  private final PWM m_pwmOutput;
-  private int m_handle;
-  private final boolean m_ownsPwm;
-
-  /**
-   * Constructs a new driver from a PWM output.
-   *
-   * @param output the pwm output to use
-   */
-  public AddressableLED(PWM output) {
-    m_pwmOutput = output;
-    m_ownsPwm = false;
-    init();
-  }
+  private final int m_pwmHandle;
+  private final int m_handle;
 
   /**
    * Constructs a new driver for a specific port.
    *
    * @param port the output port to use (Must be a PWM port)
    */
   public AddressableLED(int port) {
-    m_pwmOutput = new PWM(port);
-    m_ownsPwm = true;
-    init();
-  }
-
-  private void init() {
-    m_handle = AddressableLEDJNI.initialize(m_pwmOutput.m_handle);
+    m_pwmHandle = PWMJNI.initializePWMPort(HAL.getPort((byte) port));
```

Comment:
Need to check for port bounds:

`SensorUtil.checkPWMChannel(channel);`

Reply:
We actually don't, the HAL does that internally, and its kind of a wasted check that I never liked having anyway.

Reply:
Its just leftover from when the HAL didn't do that checking.